### ISSUE TITLE
t.web sends zero length chunk first

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7220]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#493|
|Type|defect|
|Created|2004-02-09 04:21:05Z|

Attachments:

* [soap.patch.txt](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c56/c56f5cb661da571966164b6b20c855e02fcc7044/7fabc78226f92c34b1c8cd3d7429c59f867a07bc.txt) (1457 bytes) - added by exarkun on 2004-02-09 04:21:05Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__493 493
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1076300465000000 1076300465000000
changetime__1471133134006504 1471133134006504
version__ 
owner__ 
cc__glyph cc__radix cc__exarkun cc__spiv cc__itamarst cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
Example of the result attached
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
I'm passing the potato to itamar, he is a smartieman, and I am not.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Guh. There's a empty string write first, but it's supposed to trigger writing of
headers etc.. Additionally in chunked mode empty strings should never get written.

How did you achieve this result? What client? What version HTTP, and could I see
transcript of client request?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
Oh. Boo. I was going to say that this URL consistently shows it:

http://twistedmatrix.com/users/roundup.twistd/twisted/file178/soap.patch

But apparently it's like that in Roundup's database.

I don't know how this file originally got created... but it *is* a transcript of
what is sent by the server, at least. I know I've seen this elsewhere... *maybe*
especially on CGIs? Or maybe I just browse stuff in CGIs more often (MoinMoin, etc).
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
It looks un-possible to me. Request.chunked is only set during the first call to write, in which, 
immediately afterwards, it writes out the headers (without a prefixed 0 chunk). So I can't see how can 
possibly happen.

Also, write("") won't write anything, so that couldn't result in a 0-length chunk.

The only place I can see that the 0 can come from is the finish() of a previous request on the same 
connection. But, finish() checks for being called twice, so I don't see how an extra 0 could possibly show 
up.

My current best guess without further data is that the browser screwed up somehow. How often/
recently has this happened?

Recommend this bug be closed if no further info.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
As no further info appeared, closing as unreproducible.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|GitHub commented|
|-|-|

In changeset 684d88e7b73cb4af011edc6a6a7d435508c30b99
```
#!CommitTicketReference repository="" revision="684d88e7b73cb4af011edc6a6a7d435508c30b99"
Merge pull request #493 from twisted/8754-kqueue-error-handling

Author: glyph

Reviewer: markrwilliams 

Fixes: #8754

Handle the correct exception type raised by asyncio on OS X when pipes are broken.
```
